### PR TITLE
ENH/DOC: geocentric / geodetic info for sgp4 instrument, metadata improvement

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #56, #55, #16

- Adds conversions to latitude / longitude / altitude to missions_sgp4
- Improves metadata generation and consistency throughout.
- Adds boolean kwarg 'one_orbit', which adjusts the output number of points to include a single orbit

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

By loading satellite data and plotting orbits.  Example load:
```
import pysat
sat = pysat.Instrument('missions', 'sgp4', inclination=15,
                               alt_periapsis=400, alt_apoapsis=600)
```
The output should match expectations, with max latitude corresponding to inclination, and altitude range described by periapsis and apoapsis.  Note that both spherical (eg, latitude) and elliposidal(eg, geod_latitude) calculations are included.  There remains some discrepancy on the order of 10 km in expectations.  This could be due to the `methods.convert_from_keplerian` method.

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.11
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
Putting this up as a draft pull into #60 to show downstream changes.  Currently includes lat / long / alt calculations from both spherical and ellipsoidal assumptions for comparison and testing.

**Comment by jklenzing** [comment]
Marking this as ready for review, but testing of expected orbits is required.

**Comment by JonathonMSmith** [comment]
not sure if this is because of the way I installed or something, but I only get data when I load dates between 2017-2019

**Comment by jklenzing** [comment]
> not sure if this is because of the way I installed or something, but I only get data when I load dates between 2017-2019

It's tied to how we inherit the instrument definition from the core code.  Because a file has to exist, we have a fake routine for list files.  We should be able to fix this at the instrument level by passing through a wider range of dates here:

https://github.com/pysat/pysatMissions/blob/a6121b4ebaed4b0241b0f3f7d97adc10e32c4916/pysatMissions/instruments/missions_sgp4.py#L238

**Comment by jklenzing** [comment]
Conversely, you can set a new range when initializing without changing the code:
```
sat = pysat.Instrument('missions', 'sgp4', file_date_range=pds.date_range(dt.datetime(2012, 1, 1), dt.datetime(2015, 1, 1))
```

I keep forgetting about this feature.

File: pysatMissions/instruments/missions_sgp4.py [hunk 1]
```diff
@@ -65,10 +68,10 @@ def init(self):
 clean = mcore._clean
 
 
-def load(fnames, tag=None, inst_id=None,
-         TLE1=None, TLE2=None, alt_periapsis=None, alt_apoapsis=None,
-         inclination=None, raan=None, arg_periapsis=None, mean_anomaly=None,
-         bstar=None, num_samples=None, cadence='1S'):
+def load(fnames, tag=None, inst_id=None, TLE1=None, TLE2=None,
+         alt_periapsis=None, alt_apoapsis=None,
+         inclination=None, raan=0., arg_periapsis=0., mean_anomaly=0.,
```
Comment (rstoneback) [inline pysatMissions/instruments/missions_sgp4.py:73]: I see that raan and arg_periapsis etc. are used based on kwarg settings. Seems possible a user may not know when they are used. I'd recommend adding info on the docstring to inclination that it is the key for the other parameters. Info for the other variables should be updated to reflect the new defaults.

**Review by rstoneback — COMMENTED** [review]
Thanks for the improvements @jklenzing!

**Comment by jklenzing** [comment]
Thanks @rstoneback.  I've updated docstrings and added a test for one of the new options.

**Review by rstoneback — APPROVED** [review]

**Pull request merged**